src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -15,38 +21,45 @@ def get_schema(self):
     def get_role(self):
         return self._conn.role
 
-    def get_repository_url_list(self, repo_name: str) -> SnowflakeCursor:
-        role = self.get_role()
-        database = self.get_database()
-        schema = self.get_schema()
-
-        registry_query = f"""
-            use role {role};
-            use database {database};
-            use schema {schema};
-            show image repositories like '{repo_name}';
-            """
-
-        return self._execute_query(registry_query)
+    def get_repository_row(self, repo_name: str) -> Dict:
+        if not is_valid_unquoted_identifier(repo_name):
+            raise ValueError(
+                f"repo_name '{repo_name}' is not a valid unquoted Snowflake identifier"
+            )
 
-    def get_repository_url(self, repo_name):
-        database = self.get_database()
-        schema = self.get_schema()
+        repo_name = repo_name.upper()
 
-        result_set = self.get_repository_url_list(repo_name=repo_name)
+        # because image repositories only support unquoted identifiers, SHOW LIKE should only return one or zero rows
+        repository_list_query = (
+            f"show image repositories like '{escape_like_pattern(repo_name)}'"
+        )
 
+        result_set = self._execute_schema_query(
+            repository_list_query, cursor_class=DictCursor
+        )
         results = result_set.fetchall()
+
+        colored_repo_name = click.style(f"'{repo_name}'", fg="green")
```

Comment:
Is this color needed? It only cause problems with equals check

Reply:
I think it has marginal improvement for UI but testability is probably more important. Good point.